pkg/apis/datadoghq/v1alpha1/datadogagent_types.go

```diff
@@ -476,6 +495,15 @@ type DatadogAgentSpecClusterAgentSpec struct {
 	// Cluster Agent configuration
 	Config ClusterAgentConfig `json:"config,omitempty"`
 
+	// Allow to put custom configuration for the agent, corresponding to the datadog-cluster.yaml config file
+	// +optional
+	CustomConfig *CustomConfigSpec `json:"customConfig,omitempty"`
+
+	// Confd Provide additional cluster check configurations. Each key will become a file in /conf.d
+	// see https://docs.datadoghq.com/agent/autodiscovery/ for more details.
+	// +optional
+	Confd *ConfigDirSpec `json:"confd,omitempty"`
```

Comment:
I'm wondering if it makes more sense to put these under the `Config` field  (in the `ClusterAgentConfig` struct) - same for the node agent and clc runner config

Reply:
Ok for `Confd` and `Checkd`, I will move them to the "ClusterAgentConfig" and "NodeAgentConfig" since they are used only by the corresponding container.
the CLC dont have `Confd` and `Checkd`.

For CustomConfig I will not move them since is is shared between the different containers